apps/dashboard/test/integration/dashboard_controller_test.rb

```diff
@@ -132,54 +132,56 @@ def teardown
     assert_select "nav a[href='#{batch_connect_sessions_path}']", 0
   end
 
-  test "should not create app menus if NavConfig.categories is empty and whitelist is enabled" do
+  test "should not create app menus if UserConfiguration.categories is empty and filter_nav_categories? is true" do
     SysRouter.stubs(:base_path).returns(Rails.root.join("test/fixtures/sys_with_gateway_apps"))
     OodAppkit.stubs(:clusters).returns(OodCore::Clusters.load_file("test/fixtures/config/clusters.d"))
-    NavConfig.stubs(:categories_whitelist?).returns(true)
-    NavConfig.stubs(:categories).returns([])
+    stub_user_configuration({
+      filter_nav_categories?: true,
```

Comment:
It seems like since this is computed, we wouldn't want to stub it. I.e., if we're stubbing it we're not testing it. We can stub `categories` and maybe NavConfig stuff, but I'd leave `filter_nav_categories?` alone to actually be a part of the test.

Reply:
From my point of view, we are not testing the logic for `filter_nav_categories?`, this is already tested in UserConfigurationTest. Here, the tests are about the implications for the navigation depending on the values for `categories` and `filter_nav_categories?`, that is why I thought that stubbing was ok.